tests/testthat/test-mod_review_form_tbl.R

```diff
@@ -35,102 +35,116 @@ describe(
-    #       and that a valid JSON output table will be created", {
-    #         testServer(mod_study_forms_server, args = testargs, {
-    #           ns <- session$ns
-    #           r$subject_id = "NLD_05_561"
-    #           session$setInputs(
-    #             filter = c("pulse", "BMI"),
-    #             show_all = TRUE
-    #           )
-    #           expect_true(is.data.frame(study_form_data()))
-    #           enabled_rows <- lapply(study_form_data()[["row_review_status"]], \(x) isFALSE(x$disabled)) |> unlist()
-    #           expect_equal(nrow(study_form_data()), 68)
-    #           
-    #           table_ids <- unique(study_form_data()$subject_id)
-    #           table_ids <- table_ids[order(table_ids)]
-    #           expected_ids <- unique(r$review_data$subject_id)
-    #           expected_ids <- expected_ids[order(expected_ids)]
-    #           expect_equal(table_ids, expected_ids)
-    #           
-    #           expect_true(inherits(output[["review_form_tbl-table"]], "json"))
-    #         })
-    #       })
+    it(
+      "Scenario 1 - Select row for review. Given a test [Adverse events] data set,
+        and the active subject_id set to ID 'DEU_02_482',
+        and the subject having an adverse event [Allergic Reaction] that has not yet been reviewed,
+        and setting the rows of the event [Allergic Reaction] to be selected in the form table,
+        I expect that [pending_review_records] gets updated with the approriate row numbers,
+        and that the output table is a valid JSON object.",
+      {
+        app_data <- get_appdata(clinsightful_data)
+        ae_data <- app_data[["Adverse events"]]
+        ae_rev_data <- get_review_data(ae_data) |> 
+          dplyr::mutate(id = dplyr::row_number(), reviewed = "No", status = "new")
+        testargs <- list(
+          form = "Adverse events",
+          form_data = reactiveVal(ae_data),
+          form_review_data = reactiveVal(ae_rev_data),
+          form_items = "",
+          active_subject = reactiveVal("DEU_02_482"),
+          show_all = reactiveVal(FALSE),
+          table_names = NULL,
+          title = NULL
+        ) 
+        
+        testServer(mod_review_form_tbl_server, args = testargs, {
+          ns <- session$ns
+          
+          session$userData$pending_form_review_status <- reactiveValues()
+          session$userData$pending_review_records <- reactiveValues()
+          session$userData$review_type <- reactiveVal("subject")
+          session$flushReact()
+          ae_rev_status <- merged_form_data() |> 
+            dplyr::filter(
+              subject_id == active_subject(), 
+              grepl("Allergic Reaction", Name)
+            ) |> 
+            dplyr::pull(row_review_status)
+          ae_rev_status <- ae_rev_status[[1]]
+          session$setInputs(
+            table_review_selection = data.frame(
+              id = ae_rev_status$id, 
```

Comment:
```suggestion
              id = ae_rev_status$ids, 
```
The named value is actually `ids`. R is doing a partial match here.